large-message-core/src/test/java/com/bakdata/kafka/CompressionTypeTest.java

```diff
@@ -0,0 +1,21 @@
+package com.bakdata.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Random;
+
+import org.junit.jupiter.api.Test;
+
+class CompressionTypeTest {
+    @Test
+    void shouldRoundtrip() {
+        for (CompressionType compressionType : CompressionType.values()) {
+            byte[] original = new byte[20];
+            new Random().nextBytes(original);
+            byte[] compressed = compressionType.compress(original);
+            byte[] decompressed = compressionType.decompress(compressed);
+            assertThat(decompressed.length).isEqualTo(original.length);
```

Comment:
There is most likely an assertion `assertThat(decrompessed).hasLength(...)` or `.hasSameLength(...)`